File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int finish = 0;
    int count = 0;
    
    while(finish < progresses.size()) {
        for (int i = finish; i < progresses.size(); i++) {
        
            progresses[i] += speeds[i];
        
            if (progresses[i] >= 100 && i == (finish + count))
                count++;
        
            if (i == progresses.size() - 1) {
                finish += count;
                if (count != 0)
                    answer.push_back(count);
                count = 0;
            } 
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    for i in range(0, len(speeds)):
        left = (99-progresses[i]) / speeds[i] + 1
        progresses[i] = math.floor(left)
        
    start = progresses[0]
    count = 0
    for i in range(0, len(speeds)):
        if (progresses[i] > start):
            answer.append(count)
            count = 0
            start = progresses[i]
        count += 1
    answer.append(count)
    return answer